Src/FluentAssertions/Primitives/ReferenceTypeAssertions.cs

```diff
@@ -416,6 +417,44 @@ public AndConstraint<TAssertions> NotBeAssignableTo(Type type, string because =
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the criteria provided by the element inspector is satisfied.
```

Comment:
🤔 I think that this can be phrased better. We're providing a method that is supposed to be used to execute one or more assertions on an object. And if you run multiple assertions that fail, they will be reported as one failure. We need to capture that in the name.

Reply:
How about:
>Asserts that all element inspectors for the specific instance are satisfied
I'll see if it can be aligned more closely to what is written elsewhere (this one is similar to what is written for `Match`).

Reply:
I doubt whether the term "element inspectors" will resonate well with people (even though we've used it before). 

What about 

```csharp
    /// <summary>
    /// Allows combining one or more assertions using the other assertion methods that this library offers on an instance of <typeparamref name="T"/>.
    /// </summary>
    /// <remarks>
    /// If multiple assertions executed by the <paramref name="assertion"/> fail, they will be raised as a single failure.
    /// </remarks>
    /// <param name="assertion">The element inspector which must be satisfied by the <typeparamref name="TSubject" />.</param>
    /// <returns>An <see cref="AndConstraint{T}" /> which can be used to chain assertions.</returns>
    /// <exception cref="ArgumentNullException"><paramref name="assertion"/> is <see langword="null"/>.</exception>
    public AndConstraint<TAssertions> Satisfy<T>(Action<T> assertion)
```

Reply:
Sounds good to me :+1: